nntrainer/layers/layer_node.cpp

```diff
@@ -430,7 +431,8 @@ void LayerNode::read(std::ifstream &file, bool opt_var) {
   } else {
     for (unsigned int i = 0; i < run_context->getNumWeights(); ++i) {
       /// @note shared weights are only be read at the first acecss
-      if (run_context->isGradientLastAccess(i)) {
+      if (run_context->weightHasGradient(i) &&
```

Comment:
I think this shouldn't check the weightHasGradient. we need to load the weight for the forwarding only.

Reply:
The reason to add this is that `isGradientLastAccess` is wrong or not properly set for weights that don't have a gradient.